### PR TITLE
chore: add eslint

### PR DESCRIPTION
**Pull request by Krishks369**
- Added eslint to template for ease of reading .
- removed `replaceVariablesWithValues` filter from all.js since it is present in generator-filter itself as `replaceServerVariablesWithValues`.
-  Successfully Tested the communication between mqtt client and generator locally with the changes.

### CODE REVIEW

File: filters/all.js [hunk 1]
```diff
@@ -170,8 +171,10 @@ filter.convertToFilename = convertToFilename;
  * @param {Object} serverVariables object containing server variables.
  * @return {String}
  */
+/* eslint-disable */
```
Comment (derberg) [inline filters/all.js:174]: why do we need to disable here?

Comment (derberg) [reply filters/all.js:174]: we could actually remove this filter and import it from here https://github.com/asyncapi/generator-filters/blob/master/src/customFilters.js as one guy took it one day to main filters package. You could basically just remove this function and all should work as `generator-filters` are configured here already.

please just run generation locally and test it as described in readme, to be 100% sure all works like a charm

File: package.json [hunk 2]
```diff
@@ -20,7 +20,8 @@
   "homepage": "https://github.com/asyncapi/nodejs-template#readme",
   "scripts": {
     "test": "rimraf test/temp && jest --modulePathIgnorePatterns='./template'",
-    "lint": "echo \"No linter specified yet\"",
+    "lint": "eslint --max-warnings 0 --config .eslintrc .",
+    "lint:fix": "eslint --fix --ext .js,.jsx .",
```
Comment (derberg) [inline package.json:24]: why not `eslint --fix --config .eslintrc .` ?

**Review by derberg — CHANGES_REQUESTED** [review]
few fixes suggested

**Review by Samridhi-98 — COMMENTED** [review]
@derberg  @Krishks369  I would suggest adding `package.json` and 
`package-lock.json`  in `.eslintignore`  because no point to run lint on packages.

**Review by derberg — CHANGES_REQUESTED** [review]
cool!

what left is:
- my comment to `npm`. I guess it is not needed really, so remove and refresh `package-lock.json`
- extend description that `replaceVariablesWithValues` was removed and why
- indicate that you tested locally and communication from mqtt client to generated server was successful

**Review by derberg — APPROVED** [review]
Thanks 🚀 
Awesome contribution!!!

**Pull request merged**